onnxruntime/core/framework/execution_frame.cc

```diff
@@ -184,12 +184,33 @@ void IExecutionFrame::Init(const std::vector<int>& feed_mlvalue_idxs, const std:
   // 4. handle feed in values. these can override initializer values so must be last
   for (size_t idx = 0, end = feed_mlvalue_idxs.size(); idx < end; ++idx) {
     int ort_value_idx = feed_mlvalue_idxs[idx];
-    // we are sharing the underline tensor/object for MLValue
+    // we are sharing the underlying tensor/object for MLValue
     all_values_[ort_value_idx] = feeds[idx];
   }
 }
 
-Status IExecutionFrame::GetOutputs(std::vector<OrtValue>& fetches) {
+void IExecutionFrame::UpdateFeeds(const std::vector<int>& feed_mlvalue_idxs, const std::vector<OrtValue>& feeds) {
+  for (size_t idx = 0, end = feed_mlvalue_idxs.size(); idx < end; ++idx) {
+    int ort_value_idx = feed_mlvalue_idxs[idx];
+    // we are sharing the underlying tensor/object for MLValue
+    all_values_[ort_value_idx] = feeds[idx];
+  }
+}
+
+void IExecutionFrame::UpdateFetches(const std::vector<int>& fetch_mlvalue_idxs, const std::vector<OrtValue>& fetches) {
+  if (!fetches.empty()) {
+    fetch_mlvalue_idxs_ = fetch_mlvalue_idxs;
+
+    auto num_fetches = fetch_mlvalue_idxs_.size();
+
+    for (size_t idx = 0; idx < num_fetches; ++idx) {
+      int ort_value_idx = fetch_mlvalue_idxs_[idx];
+      all_values_[ort_value_idx] = fetches[idx];
```

Comment:
>all_values_[ort_value_idx] = fetches[idx]; [](start = 6, length = 42)

similarly here, we should check the mlvalue in all_values_[ort_value_idx] hasn't been produces yet...